src/artemis/tests/test_fast_grid_scan_plan.py

```diff
@@ -60,136 +64,126 @@ def test_ispyb_params_update_from_ophyd_devices_correctly():
+@patch("artemis.fast_grid_scan_plan.move_xyz")
+@patch("artemis.fgs_communicator.wait_for_result")
+def test_results_passed_to_move_xyz(
+    wait_for_result: MagicMock, move_xyz: MagicMock, run_gridscan: MagicMock
+):
+    RE = RunEngine({})
     params = FullParameters()
-    params.grid_scan_params.z_steps = 2
-
-    FakeFGSComposite = make_fake_device(FGSComposite)
-    fgs_composite: FGSComposite = FakeFGSComposite(name="fgs", insertion_prefix="")
-
+    communicator = FGSCommunicator()
+    wait_for_result.return_value = Point3D(1, 2, 3)
+    motor_position = params.grid_scan_params.grid_position_to_motor_position(
+        Point3D(1, 2, 3)
+    )
+    FakeComposite = make_fake_device(FGSComposite)
     FakeEiger = make_fake_device(EigerDetector)
-    eiger: EigerDetector = FakeEiger(
-        detector_params=params.detector_params, name="eiger"
+    RE(
+        run_gridscan_and_move(
+            FakeComposite("test", name="fakecomposite"),
+            FakeEiger(params.detector_params),
+            params,
+            communicator,
+        )
     )
+    move_xyz.assert_called_once_with(ANY, motor_position)
 
-    return fgs_composite, eiger, params
-
-
-@patch("artemis.fast_grid_scan_plan.run_start")
-@patch("artemis.fast_grid_scan_plan.run_end")
-@patch("artemis.fast_grid_scan_plan.wait_for_result")
-@patch("artemis.fast_grid_scan_plan.StoreInIspyb3D.store_grid_scan")
-@patch("artemis.fast_grid_scan_plan.StoreInIspyb3D.get_current_time_string")
-@patch(
-    "artemis.fast_grid_scan_plan.StoreInIspyb3D.update_grid_scan_with_end_time_and_status"
-)
-def test_run_gridscan_zocalo_calls(
-    mock_ispyb_update_time_and_status: MagicMock,
-    mock_ispyb_get_time: MagicMock,
-    mock_ispyb_store_grid_scan: MagicMock,
-    wait_for_result,
-    run_end,
-    run_start,
-    dummy_3d_gridscan_args,
-):
-    dc_ids = [1, 2]
-    dcg_id = 4
-
-    mock_ispyb_store_grid_scan.return_value = [dc_ids, None, dcg_id]
-    mock_ispyb_get_time.return_value = DUMMY_TIME_STRING
-    mock_ispyb_update_time_and_status.return_value = None
 
-    print(run_start)
+@patch("bluesky.plan_stubs.mv")
+def test_results_passed_to_move_motors(bps_mv: MagicMock):
+    from artemis.fast_grid_scan_plan import move_xyz
 
-    with patch("artemis.fast_grid_scan_plan.NexusWriter"):
-        list(run_gridscan(*dummy_3d_gridscan_args))
-
-    run_start.assert_has_calls(call(x) for x in dc_ids)
-    assert run_start.call_count == len(dc_ids)
-
-    run_end.assert_has_calls(call(x) for x in dc_ids)
-    assert run_end.call_count == len(dc_ids)
-
-    wait_for_result.assert_called_once_with(dcg_id)
+    RE = RunEngine({})
+    params = FullParameters()
+    motor_position = params.grid_scan_params.grid_position_to_motor_position(
+        Point3D(1, 2, 3)
+    )
+    FakeComposite = make_fake_device(FGSComposite)
+    RE(
+        move_xyz(
+            FakeComposite("test", name="fakecomposite").sample_motors, motor_position
+        )
+    )
+    x = bps_mv
+    print(x)
```

Comment:
Nit: Remove print statement and x assignment

Reply:
done